### ISSUE TITLE
yolo2_light crashed when loading model trained by darknet.exe

### ISSUE THREAD

**Issue by goodtogood** [AlexeyAB/darknet#320]
I trained one class model by darknet.exe, cfg as follows,
after changed batch = 1  and subdivisions =1, I tried running yolo2_light.exe with trained model
It just crashed with useless  message while  darknet.exe works well
 how to find it out?
thanks!

![tiny](https://user-images.githubusercontent.com/10088733/34330918-f902f296-e95c-11e7-928b-8c2e18e47c6b.png)

> 
> 
> [net]
> batch=64
> subdivisions=8
> width=416
> height=416
> channels=3
> momentum=0.9
> decay=0.0005
> angle=0
> saturation = 1.5
> exposure = 1.5
> hue=.1
> 
> learning_rate=0.001
> max_batches = 120000
> policy=steps
> steps=-1,1000,4000,10000
> scales=.1,10,.1,.1
> 
> [convolutional]
> batch_normalize=1
> filters=16
> size=3
> stride=1
> pad=1
> activation=leaky
> 
> [maxpool]
> size=2
> stride=2
> 
> [convolutional]
> batch_normalize=1
> filters=32
> size=3
> stride=1
> pad=1
> activation=leaky
> 
> [maxpool]
> size=2
> stride=2
> 
> [convolutional]
> batch_normalize=1
> filters=64
> size=3
> stride=1
> pad=1
> activation=leaky
> 
> [maxpool]
> size=2
> stride=2
> 
> [convolutional]
> batch_normalize=1
> filters=128
> size=3
> stride=1
> pad=1
> activation=leaky
> 
> [maxpool]
> size=2
> stride=2
> 
> [convolutional]
> batch_normalize=1
> filters=256
> size=3
> stride=1
> pad=1
> activation=leaky
> 
> [maxpool]
> size=2
> stride=2
> 
> [convolutional]
> batch_normalize=1
> filters=512
> size=3
> stride=1
> pad=1
> activation=leaky
> 
> [maxpool]
> size=2
> stride=1
> 
> [convolutional]
> batch_normalize=1
> filters=1024
> size=3
> stride=1
> pad=1
> activation=leaky
> 
> ###########
> 
> [convolutional]
> batch_normalize=1
> size=3
> stride=1
> pad=1
> filters=1024
> activation=leaky
> 
> [convolutional]
> size=1
> stride=1
> pad=1
> filters=30
> activation=linear
> 
> [region]
> anchors = 0.738768,0.874946,  2.42204,2.65704,  4.30971,7.04493,  10.246,4.59428,  12.6868,11.8741
> bias_match=1
> classes=1
> coords=4
> num=5
> softmax=1
> jitter=.2
> rescore=1
> 
> object_scale=5
> noobject_scale=1
> class_scale=1
> coord_scale=1
> 
> absolute=1
> thresh = .6
> random=1
> 
>

**Comment by AlexeyAB** [comment]
@goodtogood Hi,

* Did you compile yolo_light for CPU, GPU(CUDA), or OpenCL?
* Can you run sucessfully `yolo_gpu_demo.cmd`, `yolo_gpu.cmd`, `yolo_cpu_demo.cmd`?
* What command line do you use to start detection using yolo_light?
* Try the same command line on Darknet, does it work well?
* Can you provide your 4 files: .names, .data, .cfg, .weights? For example using google disk.

**Comment by goodtogood** [comment]
Hi, @AlexeyAB
thanks for your reply !
I trained this model for people detection, just one class, currently the model is not trained well enough

1. yes, I compiled yolo_cpu && yolo_gpu successfully

2. I can run both correctly with original models yolo supplied.

3.   yolo_gpu.exe detector demo ./data/person.names tiny-yolo-person.cfg tiny-yolo-person_6000.weights  33.mp4

4. yes, same command works well on Darknet

5. yes,  [GOOGLE　DRIVER](https://drive.google.com/file/d/1Ix70vtCtpcEcR_GlMaPh6ijbN6LcWZRF/view?usp=sharing)

**Comment by AlexeyAB** [comment]
@goodtogood 

You should use: `yolo_gpu.exe detector demo person/person.names ...`
instead of `yolo_gpu.exe detector demo person/person.data...` for yolo_light

Try to use this commands:

* `yolo_gpu.exe detector demo person/person.names person/tiny-yolo-person.cfg person/tiny-yolo-person_6000.weights -thresh 0.24 33.mp4`

* `yolo_gpu.exe detector test person/person.names person/tiny-yolo-person.cfg person/tiny-yolo-person_6000.weights -thresh 0.24 dog.jpg`

**Comment by goodtogood** [comment]
sorry , it's typo, actually I used  person/**person.names**
yolo_gpu.exe stopped working and exited.

![00011111](https://user-images.githubusercontent.com/10088733/34341116-b03ffbe8-e9cb-11e7-9a35-a676bb74812f.png)

**Comment by AlexeyAB** [comment]
You should put dll-files to the `/bin` directory - where is you exe-file:
* for OpenCV:
    * if you use OpenCV 3.2.0: `opencv_world320.dll`, `opencv_ffmpeg320_64.dll`
    * if you use OpenCV 2.4.13 `opencv_core2413.dll`, `opencv_ffmpeg2413_64.dll`, `opencv_highgui2413.dll`, `opencv_imgproc2413.dll`
* if you use CUDA 8.0: `cudart64_80.dll`
* if you use cuDNN 6.0: `cudnn64_6.dll`

I just now ran this command and got good results: `yolo_gpu.exe detector demo person/person.names person/tiny-yolo-person.cfg person/tiny-yolo-person_6000.weights -thresh 0.24 test.mp4`

![image](https://user-images.githubusercontent.com/4096485/34341162-7e9ed9a2-e9a2-11e7-92b9-6d7531f14d7a.png)

**Comment by goodtogood** [comment]
I found the problem, sorry, it's my fault!  
I thought paramer thresh was optional , so sometimes I missed it .
thank you so much!!
